main/src/library/Core/System/Environment.flix

```diff
@@ -0,0 +1,61 @@
+namespace Core/System/Environment {
+  pub def getEnv(): Map[String, String] & Impure =
+    import java.lang.System:getenv();
+    import java.util.Map.entrySet();
+    import java.util.Set.iterator();
+    import java.util.Iterator.hasNext();
+    import java.util.Iterator.next();
+    import java.util.Map$Entry.getKey();
+    import java.util.Map$Entry.getValue();
+    try {
+      let iter = getenv().entrySet().iterator();
+      let iterDone : Unit ~> Bool = () -> not iter.hasNext();
+      let iterNext: Unit ~> (String, String) = () -> {
+        let entry = iter.next() as ##java.util.Map$Entry;
+        let key = entry.getKey() as String;
+        let value = entry.getValue() as String;
+        (key, value)
+      };
+      Iterator.toMap(Iterator(iterDone, iterNext))
+    } catch {
+      case _:##java.lang.Exception => Map.empty()
+    }
+
+  pub def getVar(name: String): Option[String] =
+    import java.lang.System:getenv(String);
+    try {
+      let ans = getenv(name) as & Pure;
+      if (Object.isNull(ans)) None else Some(ans)
+    } catch {
+      case _:##java.lang.Exception => None
+    }
+
+  pub def getProp(propName: String): Option[String] =
+    import java.lang.System:getProperty(String);
+    try {
+      let ans = getProperty(propName) as & Pure;
+      if (Object.isNull(ans)) None else Some(ans)
+    } catch {
+      case _:##java.lang.Exception => None
+    }
+
+  pub def getOsName(): Option[String] = getProp("os.name")
+
+  pub def getOsArch(): Option[String] = getProp("os.arch")
+
+  pub def getOsVersion(): Option[String] = getProp("os.version")
+
+  pub def getFileSeparator(): Option[String] = getProp("file.separator")
+
+  pub def getPathSeparator(): Option[String] = getProp("path.separator")
+
+  pub def getLineSeparator(): Option[String] = getProp("line.separator")
+
+  pub def getCurrentWorkingDirectory(): Option[String] = getProp("user.dir")
+
+  pub def getTemporaryDirectory(): Option[String] = getProp("java.io.tmpdir")
+
+  pub def getUserName(): Option[String] = getProp("user.name")
+
+  pub def getUserHomeDirectory(): Option[String] =  getProp("user.home")
+}
```

Comment:
I think some of these do not have to return option.

For example, `java.io.File` offers access to `pathSeparator`, and `separator` which cannot be null.

Reply:
I think at minimum that:
- getFileSeparator
- getPathSeparator
- getLineSeparator

can be implemented without returning an Option.

Reply:
I've just changed the signatures of those 3 functions